File: UI/AvaCrmUi/src/app/dtos/Commons/unit.dto.ts
export enum UnitCategory {
  General = 0,   // واحدهای عمومی
  Count = 1,     // تعداد (عدد، جین، بسته)
  Weight = 2,    // وزن (کیلوگرم، گرم، تن)
  Length = 3,    // طول (متر، سانتیمتر، کیلومتر)
  Area = 4,      // مساحت (متر مربع)
  Volume = 5,    // حجم (لیتر، میلی‌لیتر)
  Time = 6       // زمان (ساعت، روز - در صورت نیاز)
}

export interface UnitDto {
  id: number;
  name: string;
  code: string;
  category: UnitCategory;
}

export interface CreateUnitDto {
  name: string;
  code: string;
  category: UnitCategory;
}

export interface UpdateUnitDto {
  id: number;
  name: string;
  code: string;
  category: UnitCategory;
}

File: AvaCrm.Application/DTOs/Commons/Units/UnitDto.cs
using AvaCrm.Domain.Enums.Commons;
using System.ComponentModel.DataAnnotations;

namespace AvaCrm.Application.DTOs.Commons.Units;

public class UnitDto
{
    public long Id { get; set; }
    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [MaxLength(10)]
    public string Code { get; set; } = string.Empty;

    public UnitCategory Category { get; set; } = UnitCategory.General;

}
